### ISSUE TITLE
In search results to be able to see who owns the file

### ISSUE THREAD

**Issue by felisucoibi** [RetroShare/RetroShare#251]
In search results to be able to see who owns the file berofre downloading, like when you are donwloading that you can see who owns thew file, why? you can know if you have a chance or not to download it, we normally know the upload speed of our friends and how many time they are online so you can choose files owned by friends that share in good condition, if not its a lottery and you finish having lot of uncompleted files.

**Comment by realityfabric** [comment]
Wouldn't this feature undermine the anonymity granted by the file sharing visibility permissions?

You can already see which files your friends are sharing, but only if they share that information.

**Comment by TheSentry** [comment]
@realityfabric Exactly. Also, if a file is shared anonymously, this info shouldn't be available at all, otherwise this would be a big security flaw.

However, it would be interesting to see the distance to the file, i.e. how many nodes are between you and the source.

**Comment by sehraf** [comment]
i think what @felisucoibi wants is instead of just showing "_one of your friends is sharing this file_" to show "_xyz is sharing this file_". currently the search shows only how many friends share this file **when** this information is available but not who.

**Comment by realityfabric** [comment]
@TheSentry Sharing how many nodes are between you and the file would also break anonymity for files shared secretly from neighbor nodes, because you would know that the file is a neighbor node (distance == 1) and which friend is sharing the file (which friend is the file being sent from).

@sehraf If files showed who you could directly download a file from, that would be good I think. Also possibly which friends you can access a file through generally, by tunnel or by direct download, since being able to tunnel through multiple friends to get the same file also increases chances you'd be able to get the file (and also increases the chances that the speed of the download will be higher)

**Comment by realityfabric** [comment]
@TheSentry Perhaps an RTT for the file would work, but that could still lead to deanonymization.

**Comment by TheSentry** [comment]
@realityfabric Hm, I assumed that the distance information is already contained in the response to the search. I'm sure the search request contains a hop counter to limit its distance in the net. Does someone here **know** the specifics of search requests and their answers?

My point is: If this information is present in the data, not displaying it only creates a wrong sense of security/anonymity and we should think about showing it.

But I do see the information leak here, so a possible solution would be to "fake" a different distance counter for anonymously shared files, claiming they originate 1-2 hops further away.

**Comment by realityfabric** [comment]
@TheSentry iirc when you search for files it only searches up to 7 nodes deep. I'm not sure if that is handled in a way that would obscure the depth of the file from the searcher. In fact, now that I think of it, obscuring the depth from the searcher may reveal the depth of the searcher to each node searched. Not sure if that would be an issue or not.

But yes, I agree that hiding information and pretending it is not obtainable is misleading to the user, and what information is passed to where should be explicitly and carefully documented.

**Comment by felisucoibi** [comment]
Morning, like said @sehraf what we need is ONLY for direct friends, we are not interested in to know friends of friends (SHOULD BE NOT POSSIBLE FOR SECURITY REASONS), and we respect the privacy there, but, if it's a direct friend, now you just have to click "download" and  you know who owns the file, also you can check friends lists manually and try to find who is the owner, so what I ask for is to be able to see who owns it, example "this file has 3 sources, source one friend 1 source 2 friend 2 soruce 3 unknown" , WHY WE NEED IT? because if you see a file is owned only by a friend that you know upload speed is very poor, you just don't download from him.

Another approach is simple, just show owner if you see the file in your friend file list database, it's easier to search than browse all the folders manually of all your friends.

We are not interested in unhide file owners, BUT if the owners shows you his file list, why not show him in search results.

So in short, we are a community that shares files, and we trust each other, that's why we would like to see in search results more info (info that can be obtained manually, so there no reason to hide it), but only more info if they are direct friends, we respect privacy of friends of friends.

Bests

**Comment by cavebeat** [comment]
i do not want to have this "feature". 
--1;

**Comment by felisucoibi** [comment]
cavebeat probably you didn't understand... is possible to know who owns a file (if its a direct friend) just donwload it and it will show you.... so why not show it in search result.... is not a question of privacy, is a question of theres no privacy in friend owned files, ok, so lets show the info without hiding it at search results... just hide info is not secure....

**Comment by defnax** [comment]
@csoler close
